z3/snap.py

```diff
@@ -405,6 +405,17 @@ def restore(self, snap_name, dry_run=False, force=False):
         current_snap = self.s3_manager.get(snap_name)
         if current_snap is None:
             raise Exception('no such snapshot "{}"'.format(snap_name))
+        try:
+            if current_snap.key.ongoing_restore == True:
+                raise Exception('snapshot {} is currently being restore from glocier; try again later'.format(snap_name))
+            if current_snap.key.storage_class == "GLACIER":
+                current_snap.key.restore(days=5)
```

Comment:
I haven't done a restore at all, so it wasn't clear to me what I needed to do.  That would be loop that adds to to_restore?